src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -1148,6 +1156,30 @@ private void retrieveChanges(Run build, GitClient git, TaskListener listener) th
         }
     }
 
+    private boolean determineRedundantFetch(CloneOption option, RemoteConfig rc) {
+        List<RefSpec> initialFetchRefSpecs = rc.getFetchRefSpecs();
+        boolean isDefaultRefspec = true; // default refspec is any refspec with "refs/heads/" mapping
+        boolean removeSecondFetch = true;
+        if (initialFetchRefSpecs != null) {
+            for (RefSpec ref:initialFetchRefSpecs) {
+                if (!ref.toString().contains("refs/heads")) {
+                    isDefaultRefspec = false; // if refspec is not of default type, preserve second fetch
+                }
+            }
```

Comment:
If only one of the elements does not contain `refs/heads` then you are marking that is not a defaultRefspec. Could you explain what is your intention here when there is more than one element? Sorry but it's a bit unclear for me

Reply:
If honor refspec for initial clone is disabled, the first clone is going to fetch all the branches and references related to them. All of those refspecs contain "refs/heads/xx" in the <src>:<dst> format of refspec.

My assumption is that the given refspec contains ""refs/heads" this pattern, there is no need to call the second fetch in any case. 

In the case of a narrow refspec which contains any references other than of branches, we will allow the second fetch call to take place, i.e, we will not avoid the redundant fetch call. 
For an example, if honorRefSpec == false and refspec = `"+refs/heads/master:refs/remotes/origin/master +refs/pull/553/head:refs/remotes/origin/pull/553"`, we can safely assume that this is not the default refspec because we have a "refs/pull" here. 
In the case of finding multiple refspecs, if any refspec is other than the default format pattern, the code assumes that it is worth fetching with the second fetch call because it may have been missed by the first/initial clone.